R/html_dependency.R

```diff
@@ -354,8 +354,18 @@ copyDependencyToDir <- function(dependency, outputDir, mustWork = TRUE) {
   dir.create(target_dir)
   dependency$src$file <- normalizePath(target_dir, "/", TRUE)
 
-  files <- if (dependency$all_files) list.files(dir) else {
-    unlist(dependency[c('script', 'stylesheet', 'attachment')])
+  if (dependency$all_files)
+    files <- list.files(dir)
+  else {
+    file_list <- c(dependency[['script']],
+                   dependency[['stylesheet']],
+                   dependency[['attachment']])
+    if (anyUnnamed(file_list))
```

Comment:
I think it'd be clearer what's going on here if we were to have a helper function "pluck" the files from each slot first, then put them together. Something like 

```r
pluck_src <- function(x) {
  if (is.character(x)) return(x)
  if (is.list(x) && "src" %in% names(x)) return(x$src) 
  if (is.list(x)) return(pluck_src(x))
  NULL
}

files <- c(
  pluck_src(dep$script),
  pluck_src(dep$stylesheet),
  pluck_src(dep$attachment),
)
```

Reply:
The function makes sense.   I'll add that.